lib/warnings.pm

```diff
@@ -337,14 +337,22 @@ sub import
 {
     shift;
 
-    my $mask = ${^WARNING_BITS} // ($^W ? $Bits{all} : $DEFAULT) ;
-
     # append 'all' when implied (empty import list or after a lone
     # "FATAL" or "NONFATAL")
     push @_, 'all'
-	if !@_ || (@_==1 && ($_[0] eq 'FATAL' || $_[0] eq 'NONFATAL'));
-
-    ${^WARNING_BITS} = _bits($mask, @_);
+        if !@_ || (@_==1 && ($_[0] eq 'FATAL' || $_[0] eq 'NONFATAL'));
+
+    my @fatal = ();
+    foreach my $warning (@_) {
+        if($warning =~ /^(NON)?FATAL$/) {
+            @fatal = ($warning);
```

Comment:
This seems to be introducing new behavior where `(NON)?FATAL` will get recognized in places other than the beginning of the import list.

Reply:
Looking closer at this, I see that it is because `_bits` is now called individually for each argument. I am not perfectly confident in the robustness of this logic but it seems correct.

Reply:
FATAL is already recognised other than just at the beginning. For example, in 5.30.3 ...

```
$ perl -E 'use warnings qw(once FATAL redefine); $once=""; $SIG{__DIE__} = sub { print "Fatal\n" }; sub foo {} *foo = sub {0}'
Name "main::once" used only once: possible typo at -e line 1.
Fatal
Subroutine main::foo redefined at -e line 1.
$ echo $?
255

$ perl -E 'use warnings qw(once redefine); $once=""; $SIG{__DIE__} = sub { print "Fatal\n" }; sub foo {} *foo = sub {0}'
Name "main::once" used only once: possible typo at -e line 1.
Subroutine main::foo redefined at -e line 1.
$ echo $?
0
```

This behaviour is unchanged in this branch.